routes/articles.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+
+const express = require('express');
+const router = express.Router();
+const knex = require('../db');
+const validate = require('../middleware/validation');
+
+const articleInfo = {
+  articles: undefined,
+}
+
+router.get('/', (req, res) => {
+  knex('articles')
+    .select('url_title', 'title', 'author', 'body')
+    .then((selection) => {
+      console.log(selection);
+      articleInfo.articles = selection;
+      res.render('articles/index', articleInfo)
+    })
+});
+
+router.get('/new', (req, res) => {
+  res.render('articles/new');
+});
+
+router.get('/:url_title', (req, res) => {
+  let url_title = req.params.url_title;
+  
+  knex('articles')
+    .select('url_title', 'title', 'author', 'body', 'created_at', 'updated_at')
+    .where('url_title', url_title)
+    .then((selection) => {
+      console.log(selection);
+      articleInfo.articles = selection[0];
+      res.render('articles/article', articleInfo);
+    });
+});
+
+router.get('/:url_title/edit?', (req, res) => {
+  let url_title = req.params.url_title;
+
+  knex('articles')
+    .select('url_title', 'title', 'author', 'body', 'updated_at')
+    .where('url_title', url_title)
+    .then((selection) => {
+      articleInfo.articles = selection[0];
+      res.render('articles/edit', articleInfo);
+    });
+});
+
+router.post('/', validate.articles, (req, res) => {
+  const data = req.body;
+
+  knex('articles')
+    .insert({
+      title: data.title,
+      url_title: encodeURI(data.title),
+      author: data.author,
+      body: data.body
+    })
+    .then(
+      knex
```

Comment:
It doesn't look like you do anything with the data you get from the db here. It's best to remove it then.